From c4d6375e61dcdfdda8fdbfbacbb1a36353d22c0c Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Wed, 08 Aug 2018 06:55:02 +0300
Subject: [PATCH] Add BTL support Fixup [specs preperation]

Now we can include specs repository headers and
use from it any register definition as is
Currently specs/dali/asic_reg is available
---

diff --git a/board/habanalabs/dali/btl/Makefile b/board/habanalabs/dali/btl/Makefile
index 66399f4..96e59d3 100644
--- a/board/habanalabs/dali/btl/Makefile
+++ b/board/habanalabs/dali/btl/Makefile
@@ -7,6 +7,9 @@
 COMMON_CPP_FLAGS += -DPREBOOT_ADDR=0x7FF0400000
 COMMON_CPP_FLAGS += -DPREBOOT_ADDR_H=0x7F
 COMMON_CPP_FLAGS += -DPREBOOT_ADDR_L=0xF0400000
+
+COMMON_CPP_FLAGS += -I$(dir $(BR2_EXTERNAL))/specs/dali/asic_reg
+
 ifeq ($(DALI_TARGET),pldm)
 	COMMON_CPP_FLAGS += -DTARGET_IS_PLDM
 endif
diff --git a/board/habanalabs/dali/btl/btl-as.py b/board/habanalabs/dali/btl/btl-as.py
index 4c4720a..20e5c47 100755
--- a/board/habanalabs/dali/btl/btl-as.py
+++ b/board/habanalabs/dali/btl/btl-as.py
@@ -33,8 +33,8 @@
     global command_menemonic
 
     addr, data = params.split(",", 2)
-    __opcode_addr = struct.pack('>I',(_opcode << 28) + int(addr,16))
-    __data = struct.pack('>I', int(data,16))
+    __opcode_addr = struct.pack('>I',(_opcode << 28) + (0xC << 24) + int(addr,16))
+    __data = struct.pack('>I', int(data,0))
     crc = binascii.crc32(__opcode_addr)
     crc = binascii.crc32(__data, crc) & 0xffffffff
     __crc = struct.pack('<I', crc)
diff --git a/board/habanalabs/dali/btl/i2c.s b/board/habanalabs/dali/btl/i2c.s
index 4b1fbbe..8a1ae2f 100644
--- a/board/habanalabs/dali/btl/i2c.s
+++ b/board/habanalabs/dali/btl/i2c.s
@@ -3,20 +3,20 @@
  */
 
 /* Program IC_SAR with address to which DW_apb_i2c slave responds */
-cfg 0xCC42008, 0x47
+cfg 0xC42008, 0x47
 
 /* Program IC_CON */
-cfg 0xCC42000, 0x22
+cfg 0xC42000, 0x22
 
 /* Write to IC_INTR_MASK to enable all interrupts */
-cfg 0xCC4202C, 0x7FFF
+cfg 0xC4202C, 0x7FFF
 
 /* Write to IC_RX_TL to set Rx FIFO threshold level */
-cfg 0xCC42038, 0
+cfg 0xC42038, 0
 
 /* Write to IC_TX_TL to set Tx FIFO threshold level */
-cfg 0xCC4203C, 0x7
+cfg 0xC4203C, 0x7
 
 /* Write 1 to IC_ENABLE to enable DW_apb_i2c */
-cfg 0xCC4206C, 0x1
+cfg 0xC4206C, 0x1
 
diff --git a/board/habanalabs/dali/btl/pcie.S b/board/habanalabs/dali/btl/pcie.S
index 417958c..678a634 100644
--- a/board/habanalabs/dali/btl/pcie.S
+++ b/board/habanalabs/dali/btl/pcie.S
@@ -2,14 +2,8 @@
  * This file is for PCIe controller initialization
  */
 
-#ifdef PCIE_BAR4_4G
-cfg 0xCC07490, 0x80000000	/* PCIE_AUX_DBI		*/
-cfg 0xCC02020, 0xFFFFFFFC	/* PCIE_DBI_BAR4_REG	*/
-cfg 0xCC07490, 0x0		/* PCIE_AUX_DBI		*/
-#endif
-
 #ifdef TARGET_IS_PLDM
-cfg 0xCC07428, 0x1		/* LTSSM_EN		*/
+cfg 0xC07428, 0x1		/* LTSSM_EN */
 #endif
 
 end
diff --git a/board/habanalabs/dali/btl/pcie_phy.s b/board/habanalabs/dali/btl/pcie_phy.s
index 5fc8dd9..a9f15ec 100644
--- a/board/habanalabs/dali/btl/pcie_phy.s
+++ b/board/habanalabs/dali/btl/pcie_phy.s
@@ -7,49 +7,49 @@
  * Set PCIe PHY CR Para I/F to CFG.
  * Select CR PARA I/F instead of JTAG
  */
-cfg 0xCC10850, 0x00000001
-cfg 0xCC10854, 0x00000001
-cfg 0xCC10858, 0x00000001
-cfg 0xCC1085C, 0x00000001
+cfg 0xC10850, 0x00000001
+cfg 0xC10854, 0x00000001
+cfg 0xC10858, 0x00000001
+cfg 0xC1085C, 0x00000001
 
 /* Select indirect access to CR PARA I/F instead of PRBI */
-cfg 0xCC10920, 0x00000001
+cfg 0xC10920, 0x00000001
 
 /* Errata: disable unused design that cause PCIe phy SRAM DERR */
-cfg 0xCC10840, 0x0
+cfg 0xC10840, 0x0
 #endif
 
 /* Adjust the incorrect PHY protocol0 defaults */
 /* writel(0xA067, mmPCIE_PHY_PROTOCOL0_EXT_MPLLA_BANDWIDTH); */
-cfg 0xCC1010C, 0x0000A067
+cfg 0xC1010C, 0x0000A067
 /* writel(0x19, mmPCIE_PHY_PROTOCOL0_EXT_MPLLA_MULTIPLIER); */
-cfg 0xCC10128, 0x00000019
+cfg 0xC10128, 0x00000019
 /* writel(0xA01E, mmPCIE_PHY_PROTOCOL0_EXT_MPLLB_BANDWIDTH); */
-cfg 0xCC10140, 0x0000A01E
+cfg 0xC10140, 0x0000A01E
 /* writel(0x50, mmPCIE_PHY_PROTOCOL0_EXT_MPLLB_MULTIPLIER); */
-cfg 0xCC10158, 0x00000050
+cfg 0xC10158, 0x00000050
 /* writel(0xE, mmPCIE_PHY_PROTOCOL0_EXT_RX_MISC_G1_SLICE0); */
-cfg 0xCC1029C, 0x0000000E
+cfg 0xC1029C, 0x0000000E
 /* writel(0xC, mmPCIE_PHY_PROTOCOL0_EXT_RX_MISC_G2_SLICE0); */
-cfg 0xCC102A4, 0x0000000C
+cfg 0xC102A4, 0x0000000C
 /* writel(0xA, mmPCIE_PHY_PROTOCOL0_EXT_RX_MISC_G3_SLICE0); */
-cfg 0xCC102AC, 0x0000000A
+cfg 0xC102AC, 0x0000000A
 /* writel(0x9, mmPCIE_PHY_PROTOCOL0_EXT_RX_MISC_G4_SLICE0); */
-cfg 0xCC102B4, 0x00000009
+cfg 0xC102B4, 0x00000009
 /* writel(0x4, mmPCIE_PHY_PROTOCOL0_EXT_TX_VBOOST_LVL); */
-cfg 0xCC10364, 0x00000004
+cfg 0xC10364, 0x00000004
 /* writel(0x1, mmPCIE_PHY_PHY_EXT_CTRL_SEL); */
-cfg 0xCC10640, 0x00000001
+cfg 0xC10640, 0x00000001
 
 /* Clear phy_init (removes the workaround provided by SNPS) */
-cfg 0xCC07100, 0x00000000
+cfg 0xC07100, 0x00000000
 
 /* Take PCIe PHY out of reset */
-cfg 0xCC01300, 0x00000001
+cfg 0xC01300, 0x00000001
 
 #ifdef PRSTN
 /* Enable PCIe FW download FSM */
-cfg 0xCC4B020, 0x00000001
+cfg 0xC4B020, 0x00000001
 #endif
 
 #ifndef TARGET_IS_PLDM
diff --git a/board/habanalabs/dali/btl/pll_pci.s b/board/habanalabs/dali/btl/pll_pci.s
index de70182..567e198 100644
--- a/board/habanalabs/dali/btl/pll_pci.s
+++ b/board/habanalabs/dali/btl/pll_pci.s
@@ -1,30 +1,30 @@
 /* PLL PSOC PCIe setup */
 
-cfg 0xCC72134, 0x1	/* WREG32(mmPSOC_PCI_PLL_RST, 1); */
-cfg 0xCC72100, 0x4	/* WREG32(mmPSOC_PCI_PLL_NR, 4); */
-cfg 0xCC72104, 0x157	/* WREG32(mmPSOC_PCI_PLL_NF, 343); */
-cfg 0xCC72108, 0x3	/* WREG32(mmPSOC_PCI_PLL_OD, 3); */
-cfg 0xCC7210C, 0xAB 	/* WREG32(mmPSOC_PCI_PLL_NB, 171); */
-cfg 0xCC72130, 0x11	/* WREG32(mmPSOC_PCI_PLL_DATA_CHNG, 0x11); */
+cfg 0xC72134, 0x1	/* WREG32(mmPSOC_PCI_PLL_RST, 1); */
+cfg 0xC72100, 0x4	/* WREG32(mmPSOC_PCI_PLL_NR, 4); */
+cfg 0xC72104, 0x157	/* WREG32(mmPSOC_PCI_PLL_NF, 343); */
+cfg 0xC72108, 0x3	/* WREG32(mmPSOC_PCI_PLL_OD, 3); */
+cfg 0xC7210C, 0xAB 	/* WREG32(mmPSOC_PCI_PLL_NB, 171); */
+cfg 0xC72130, 0x11	/* WREG32(mmPSOC_PCI_PLL_DATA_CHNG, 0x11); */
 
 /* delay before taking PLL out of reset */
 #include "nop.s"	/* udelay(100); */
 
-cfg 0xCC72134, 0	/* WREG32(mmPSOC_PCI_PLL_RST, 0); */
-cfg 0xCC722A0, 0x1	/* WREG32(mmPSOC_PCI_PLL_DIV_EN_0, 0x1); */
-cfg 0xCC72280, 0x1	/* WREG32(mmPSOC_PCI_PLL_DIV_SEL_0, 0x1); */
+cfg 0xC72134, 0	/* WREG32(mmPSOC_PCI_PLL_RST, 0); */
+cfg 0xC722A0, 0x1	/* WREG32(mmPSOC_PCI_PLL_DIV_EN_0, 0x1); */
+cfg 0xC72280, 0x1	/* WREG32(mmPSOC_PCI_PLL_DIV_SEL_0, 0x1); */
 
-cfg 0xCC72204, 0x4	/* WREG32(mmPSOC_PCI_PLL_DIV_FACTOR_1, 0x4); */
-cfg 0xCC72224, 0x1	/* WREG32(mmPSOC_PCI_PLL_DIV_FACTOR_CMD_1, 0x1); */
-cfg 0xCC722A4, 0x1	/* WREG32(mmPSOC_PCI_PLL_DIV_EN_1, 0x1); */
-cfg 0xCC72284, 0x3	/* WREG32(mmPSOC_PCI_PLL_DIV_SEL_1, 0x3); */
+cfg 0xC72204, 0x4	/* WREG32(mmPSOC_PCI_PLL_DIV_FACTOR_1, 0x4); */
+cfg 0xC72224, 0x1	/* WREG32(mmPSOC_PCI_PLL_DIV_FACTOR_CMD_1, 0x1); */
+cfg 0xC722A4, 0x1	/* WREG32(mmPSOC_PCI_PLL_DIV_EN_1, 0x1); */
+cfg 0xC72284, 0x3	/* WREG32(mmPSOC_PCI_PLL_DIV_SEL_1, 0x3); */
 
-cfg 0xCC72208, 0x8	/* WREG32(mmPSOC_PCI_PLL_DIV_FACTOR_2, 0x8); */
-cfg 0xCC72228, 0x1	/* WREG32(mmPSOC_PCI_PLL_DIV_FACTOR_CMD_2, 0x1); */
-cfg 0xCC722A8, 0x1	/* WREG32(mmPSOC_PCI_PLL_DIV_EN_2, 0x1); */
-cfg 0xCC72288, 0x3	/* WREG32(mmPSOC_PCI_PLL_DIV_SEL_2, 0x3); */
+cfg 0xC72208, 0x8	/* WREG32(mmPSOC_PCI_PLL_DIV_FACTOR_2, 0x8); */
+cfg 0xC72228, 0x1	/* WREG32(mmPSOC_PCI_PLL_DIV_FACTOR_CMD_2, 0x1); */
+cfg 0xC722A8, 0x1	/* WREG32(mmPSOC_PCI_PLL_DIV_EN_2, 0x1); */
+cfg 0xC72288, 0x3	/* WREG32(mmPSOC_PCI_PLL_DIV_SEL_2, 0x3); */
 
-cfg 0xCC7220C, 0x55	/* WREG32(mmPSOC_PCI_PLL_DIV_FACTOR_3, 0x55); */
-cfg 0xCC7222C, 0x1	/* WREG32(mmPSOC_PCI_PLL_DIV_FACTOR_CMD_3, 0x1); */
-cfg 0xCC722AC, 0x1	/* WREG32(mmPSOC_PCI_PLL_DIV_EN_3, 0x1); */
-cfg 0xCC7228C, 0x3	/* WREG32(mmPSOC_PCI_PLL_DIV_SEL_3, 0x3); */
+cfg 0xC7220C, 0x55	/* WREG32(mmPSOC_PCI_PLL_DIV_FACTOR_3, 0x55); */
+cfg 0xC7222C, 0x1	/* WREG32(mmPSOC_PCI_PLL_DIV_FACTOR_CMD_3, 0x1); */
+cfg 0xC722AC, 0x1	/* WREG32(mmPSOC_PCI_PLL_DIV_EN_3, 0x1); */
+cfg 0xC7228C, 0x3	/* WREG32(mmPSOC_PCI_PLL_DIV_SEL_3, 0x3); */
diff --git a/board/habanalabs/dali/btl/primary.S b/board/habanalabs/dali/btl/primary.S
index 1d3276a..05d8241 100644
--- a/board/habanalabs/dali/btl/primary.S
+++ b/board/habanalabs/dali/btl/primary.S
@@ -7,16 +7,16 @@
 nop				/* DUMMY */
 
 /* CPU_BOOT_STATUS_IN_BTL -> mmPSOC_GLOBAL_CONF_WARM_REBOOT */
-cfg 0xCC4B300, 0x4
+cfg 0xC4B300, 0x4
 
 /* To avoid false prstn configurae pad as schmitt trigger */
-cfg 0xCC4B668, 0x41
+cfg 0xC4B668, 0x41
 
 /* Configure prstn to reset entire SoC */
-cfg 0xCC4B470, 0x7FFFFF
+cfg 0xC4B470, 0x7FFFFF
 
 /* Clear masking of prstn from SW reset */
-cfg 0xCC4B484, 0x0
+cfg 0xC4B484, 0x0
 
 #include "pcie_phy.s"
 
@@ -38,18 +38,18 @@
  * 30 - UBOOT_OFFSET
  * 31 - UBOOT_SIZE
  */
-cfg 0xCC4B16C, BTLID
-cfg 0xCC4B170, SPL_OFFSET
-cfg 0xCC4B174, SPL_SIZE
-cfg 0xCC4B178, UBOOT_OFFSET
-cfg 0xCC4B17C, UBOOT_SIZE
+cfg 0xC4B16C, BTLID
+cfg 0xC4B170, SPL_OFFSET
+cfg 0xC4B174, SPL_SIZE
+cfg 0xC4B178, UBOOT_OFFSET
+cfg 0xC4B17C, UBOOT_SIZE
 
 /* Set CPU0 Reset address to PREBOOT */
-cfg 0xC441104, PREBOOT_ADDR_L
-cfg 0xC441114, PREBOOT_ADDR_H
+cfg 0x441104, PREBOOT_ADDR_L
+cfg 0x441114, PREBOOT_ADDR_H
 
 /* Wake up CPU0, after we put all CPUs in reset */
-cfg 0xC441124, 0x0
-cfg 0xC441124, 0x10111
+cfg 0x441124, 0x0
+cfg 0x441124, 0x10111
 
 end
diff --git a/board/habanalabs/dali/btl/uart.s b/board/habanalabs/dali/btl/uart.s
index 3369599..f5ecfb9 100644
--- a/board/habanalabs/dali/btl/uart.s
+++ b/board/habanalabs/dali/btl/uart.s
@@ -9,25 +9,18 @@
 #include "pll_pci.s"
 #endif
 
-cfg 0xCC4B9C0, 0x1	/* GPIO48 [UART0 RXD] */
-cfg 0xCC4B9C4, 0x1	/* GPIO49 [UART0 TXD] */
-cfg 0xCC45008, 0x0	/* UART0  [CTRL]      */
+cfg 0xC4B9C0, 0x1	/* GPIO48 [UART0 RXD] */
+cfg 0xC4B9C4, 0x1	/* GPIO49 [UART0 TXD] */
+cfg 0xC45008, 0x0	/* UART0  [CTRL]      */
 #ifdef TARGET_IS_PLDM
-cfg 0xCC45010, 0x17	/* UART0  [BAUDDIV]   */
+cfg 0xC45010, 0x17	/* UART0  [BAUDDIV]   */
 #elif TARGET_IS_ASIC
-cfg 0xCC45010, 0x1B2	/* UART0  [BAUDDIV]   */
+cfg 0xC45010, 0x1B2	/* UART0  [BAUDDIV]   */
 #else
-cfg 0xCC45010, 0x364	/* UART0  [BAUDDIV]   */
+cfg 0xC45010, 0x364	/* UART0  [BAUDDIV]   */
 #endif
-cfg 0xCC45008, 0x3	/* UART0  [CTRL]      */
+cfg 0xC45008, 0x3	/* UART0  [CTRL]      */
 
-/* printf ("BTL %d\n\r", is_primary ? 1 : 2)  */
-#ifdef TARGET_IS_PLDM
-cfg 0xCC45000, 0X42	/* UART0  [DATA]      */
-cfg 0xCC45000, 0X54	/* UART0  [DATA]      */
-cfg 0xCC45000, 0X4C	/* UART0  [DATA]      */
-cfg 0xCC45000, 0X20	/* UART0  [DATA]      */
-#endif
-cfg 0xCC45000, BTLID	/* UART0  [DATA]      */
-cfg 0xCC45000, 0X0A	/* UART0  [DATA]      */
-cfg 0xCC45000, 0X0D	/* UART0  [DATA]      */
+cfg 0xC45000, BTLID	/* UART0  [DATA]      */
+cfg 0xC45000, 0X0A	/* UART0  [DATA]      */
+cfg 0xC45000, 0X0D	/* UART0  [DATA]      */
